nilearn/datasets/func.py

```diff
@@ -441,7 +441,7 @@ def fetch_adhd(n_subjects=None, data_dir=None, url=None, resume=True,
     nitrc_ids = range(7782, 7822)
     max_subjects = len(ids)
     if n_subjects is None:
-        n_subjects = max_subjects
+        n_subjects = 30
```

Comment:
This seems to be the essential change of this PR (along with the doctrings changes above). I would only keep those and create an extra PR for the pep8.